include/maxminddb.h

```diff
@@ -161,6 +161,7 @@ typedef struct MMDB_ipv4_start_node_s {
 } MMDB_ipv4_start_node_s;
 
 typedef struct MMDB_s {
+    bool file_mapping;
```

Comment:
Another option than using flags is declaring a:

    static const char* memory_filename_placeholder = "[memory]";

In maxminddb.c and setting filename to it in `MMDB_open_memory`, then comparing the pointer (not the string) in `free_mmdb_struct`.